File: services/internal-treasury/src/main/java/br/com/deltaglobalbank/internal_treasury/features/listInternalTrasference/ListInternalTransferenceUseCase.java
package br.com.deltaglobalbank.internal_treasury.features.listInternalTrasference;

import br.com.deltaglobalbank.internal_treasury.domain.internalTransference.InternalTransference;
import br.com.deltaglobalbank.internal_treasury.domain.internalTransference.InternalTransferenceRepository;
import br.com.deltaglobalbank.internal_treasury.domain.shared.PaymentsStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class ListInternalTransferenceUseCase {

    private final InternalTransferenceRepository repository;

    public ListInternalTransferenceUseCase(InternalTransferenceRepository repository) {
        this.repository = repository;
    }

    public ListInternalTransferenceResponse execute(int page, int pageSize, String filter) {
        Pageable pageable = PageRequest.of(page, pageSize);

        Page<InternalTransference> result;
        if (filter.equals("all")) {
            result = repository.findAll(pageable);
        } else {
            PaymentsStatus status = PaymentsStatus.valueOf(filter.toUpperCase());
            result = repository.findByStatus(status, pageable);
        }

        List<InternalTransferenceItem> content = result.getContent().stream()
            .map(it -> new InternalTransferenceItem(
                it.id(),
                it.payerId(),
                it.accountNumber(),
                it.amount(),
                it.status().name(),
                it.requestedAt()
            ))
            .toList();

        return new ListInternalTransferenceResponse(
            content,
            result.getNumber(),
            result.getSize(),
            result.getTotalElements(),
            result.getTotalPages()
        );
    }
}

File: services/internal-treasury/src/main/kotlin/br/com/deltaglobalbank/internal_treasury/features/listInternalTrasference/ListInternalTransferenceUseCase.kt
package br.com.deltaglobalbank.internal_treasury.features.listInternalTrasference

import br.com.deltaglobalbank.internal_treasury.domain.internalTransference.InternalTransferenceRepository
import br.com.deltaglobalbank.internal_treasury.domain.shared.PaymentsStatus
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service

@Service
class ListInternalTransferenceUseCase(
    private val repository: InternalTransferenceRepository
) {
    fun execute(page: Int, pageSize: Int, filter: String): ListInternalTransferenceResponse {
        val pageable = PageRequest.of(page, pageSize)

        val result = if (filter == "all") {
            repository.findAll(pageable)
        } else {
            val status = PaymentsStatus.valueOf(filter.uppercase())
            repository.findByStatus(status, pageable)
        }

        return ListInternalTransferenceResponse(
            content = result.content.map {
                InternalTransferenceItem(
                    id = it.id,
                    payerId = it.payerId,
                    accountNumber = it.accountNumber,
                    amount = it.amount,
                    status = it.status.name,
                    requestedAt = it.requestedAt
                )
            },
            page = result.number,
            pageSize = result.size,
            totalItems = result.totalElements,
            totalPages = result.totalPages
        )
    }
}